Generator/Strategy.ts

```diff
@@ -32,11 +38,25 @@ export class Strategy{
         this.cells = cells;
         this.identified = false;
         this.values = new Array();
+        this.notes = new Array();
+    }
+
+    /**
+     * Gets cells that "cause" strategy to be applicable
+     * @returns cells
+     * @throws {@link CustomError}
+     * Thrown if strategy hasn't been identified
+     */
+    public getCause():Cell[][] {
+        if (!this.identified) {
```

Comment:
If this if statement is needed for each function in this class, can be made into its own function and called. Is this if statement needed for isNakedSingle or getStradegyType?